class_defines.py

```diff
@@ -2345,35 +2358,72 @@ def matrix_basis(self):
             # TODO: Support distance in 3d
             return Matrix()
 
-        e1, e2 = self.entity1, self.entity2
-        if e1.is_line():
-            e1, e2 = e1.p1, e1.p2
-
         sketch = self.sketch
         x_axis = Vector((1, 0))
         alignment = self.align
         align = alignment != "NONE"
 
-        if type(e2) in point_2d:
-            p1, p2 = e1.co, e2.co
-            if align:
-                v_rotation = Vector((1.0, 0.0)) if alignment == "HORIZONTAL" else Vector((0.0, 1.0))
+        e1, e2 = self.entity1, self.entity2
+        #   e1       e2
+        #   ----------------
+        #   line     [none]
+        #   point    point
+        #   point    line
+        #   arc      point
+        #   arc      line
+        #   circle   point
+        #   circle   line
+
+        # set p1 and p2
+        if e1.is_curve():
+            # reframe as point->point and continue
+            centerpoint = e1.ct.co
+            if e2.is_line():
+                p2, whatever = intersect_point_line(centerpoint, e2.p1.co, e2.p2.co)
             else:
-                v_rotation = p2 - p1
-            v_translation = (p2 + p1) / 2
+                assert(isinstance(e2, SlvsPoint2D))
+                p2 = e2.co
+            vec = (p2 - centerpoint) / (p2 - centerpoint).length
+            p1 = centerpoint + (e1.radius * Vector(vec))
+        elif e1.is_line():
+            # reframe as point->point and continue
+            e1, e2 = e1.p1, e1.p2
+            p1, p2 = e1.co, e2.co
+        else:
+            assert(isinstance(e1,SlvsPoint2D))
+            p1 = e1.co
+
 
+        # NOTE: p2 is still not defined if e1 is point
+        if type(e2) in point_2d:
+        # this includes "Line Length" (became Point->Point)
+        # and Curve -> Point
+            if not align:
+                v_rotation = p2 - p1
+            else:
+                v_rotation = Vector((1.0, 0.0)) if alignment == "HORIZONTAL" else Vector((0.0, 1.0))
             angle = v_rotation.angle_signed(x_axis)
             mat_rot = Matrix.Rotation(angle, 2, "Z")
+            v_translation = (p2 + p1) / 2
 
-        elif isinstance(e2, SlvsLine2D):
-            line = e2
-            orig = line.p1.co
-            end = line.p2.co - orig
-            angle = functions.range_2pi(math.atan2(end[1], end[0])) + math.pi / 2
-
-            mat_rot = Matrix.Rotation(angle, 2, "Z")
-            p1 = e1.co - orig
-            v_translation = (p1 + p1.project(end)) / 2 + orig
+        elif e2.is_line():
+            if e1.is_curve():
+                if not align:
+                    v_rotation = p2 - p1
```

Comment:
This vector could be zero, that case should be handled somehow otherwise we run into errors later

Reply:
fixed